src/sql/cmd/connect.py

```diff
@@ -0,0 +1,53 @@
+try:
+    from jupysql_plugin.widgets import ConnectorWidget
+except ModuleNotFoundError:
+    ConnectorWidget = None
+
+from ploomber_core.dependencies import requires
+from configparser import ConfigParser
+import json
+
+
+@requires(["jupysql_plugin"])
+def connect(others):
+    """
+    Implementation of `%sqlcmd connect`
+    """
+
+    connectorwidget = ConnectorWidget()
+
+    config = get_connections_config_file()
+    sections = config.sections()
+    update_connectorwidget_connect(
+        sections[-1], config[sections[-1]]["drivername"], connectorwidget
+    )
+    update_connectorwidget_stored_conn(config, connectorwidget)
+
+    return connectorwidget
+
+
+def get_connections_config_file():
+    config = ConfigParser()
+    config.read("connections.ini")
+    return config
+
+
+def update_connectorwidget_connect(alias, drivername, CONNECTORWIDGET):
+    try:
+        CONNECTORWIDGET._connect({"name": alias, "driver": drivername})
+        CONNECTORWIDGET.send({"method": "connected", "message": alias})
+    except Exception as e:
+        CONNECTORWIDGET.send_error_message_to_front(e)
+
+
+def update_connectorwidget_stored_conn(current_config, CONNECTORWIDGET):
+    sections = current_config.sections()
+    if len(sections) > 0:
+        connections = [
+            {"name": s, "driver": current_config[s]["drivername"]} for s in sections
+        ]
+
+        CONNECTORWIDGET.stored_connections = connections
+        CONNECTORWIDGET.send(
+            {"method": "update_connections", "message": json.dumps(connections)}
+        )
```

Comment:
I'm confused about all of this logic. this PR should limit to call `ConnectorWidget()`, it should not attempt to fix any logic or add any new features.

once it's integrated, we can fix any issues with ConnectorWidget, but we should do it in the jupysql-plugin repository, not here

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/783/files#diff-97c7661adb8698813af7c891197f6961ea293d1d1b02be2d5928d07db813bbe2R1-R16)

I was overthinking. It doesn't need `update_connectorwidget_connect` and `update_connectorwidget_stored_conn`. These two functions are deleted now.